executor/memtable_reader.go

```diff
@@ -997,3 +997,116 @@ func (e *hotRegionsHistoryRetriver) getHotRegionRowWithSchemaInfo(
 	}
 	return row, nil
 }
+
+type tikvRegionPeersRetriever struct {
+	dummyCloser
+	extractor *plannercore.TikvRegionPeersExtractor
+	retrieved bool
+}
+
+func (e *tikvRegionPeersRetriever) retrieve(ctx context.Context, sctx sessionctx.Context) ([][]types.Datum, error) {
+	if e.extractor.SkipRequest || e.retrieved {
+		return nil, nil
+	}
+	e.retrieved = true
+	tikvStore, ok := sctx.GetStore().(helper.Storage)
+	if !ok {
+		return nil, errors.New("Information about hot region can be gotten only when the storage is TiKV")
+	}
+	tikvHelper := &helper.Helper{
+		Store:       tikvStore,
+		RegionCache: tikvStore.GetRegionCache(),
+	}
+
+	if len(e.extractor.StoreIDs) < 1 && len(e.extractor.RegionIDs) < 1 {
+		regionsInfo, err := tikvHelper.GetRegionsInfo()
+		if err != nil {
+			return nil, err
+		}
+		return e.packTiKVRegionPeersRows(regionsInfo.Regions)
+	}
+
+	var regionsInfo, regionsInfoByStoreID, regionsInfoByRegionID []helper.RegionInfo
+	regionMap := make(map[int64]bool)
```

Comment:
I suggest using the empty struct to do this.

```suggestion
	regionMap := make(map[int64]struct{})
```